### DISCUSSION TITLE
Configuration property to disable DGS

### DISCUSSION THREAD

**Discussion by EugenMayer** [Netflix/dgs-framework#514] [Ideas]
When introducing DGS or when running in different profiles, it would be very handy to be able to disable DGS using a property.

It is not possible to exclude `DgsAutoConfiguration`,`DgsWebMvcAutoConfiguration` using spring.autoconfigure.exclude since this would still include the rest  controller and fail to start due to `DgsWebMvcConfigurationProperties` failing.

It would be helpful to have something like flyway offers with

```
@ConditionalOnProperty(prefix = "spring.flyway", name = "enabled", matchIfMissing = true)
```
on all the required Components. Similar to the existing 

```
@ConditionalOnProperty(name = ["dgs.graphql.graphiql.enabled"], havingValue = "true", matchIfMissing = true)
```

But for the entire DGS framework. Any thoughts why this is a bad idea?

**Comment by paulbakker** [comment]
Who would _ever_ want to disable the DGS framework!? ;-)

More seriously, yes, this is a good suggestion. Could you create an issue as a feature request for this?

**Reply by EugenMayer** [reply]
Sure I will, thank you for thinking over it :)

**Reply by EugenMayer** [reply]
Since I created the FR the thread can be closed, if that is actually a thing or desired :)
